source/_posts/2015-07-23-easier-life-with-xcode-marks-while-using-bash-scripts.markdown

```diff
@@ -0,0 +1,35 @@
+---
+layout: post
+title: "Easier life with XCode marks while using Bash scripts"
+date: 2015-07-23 13:24:06 +0200
+comments: false
+author: kondrat
+categories:
+---
+Bash scripts are commonly used in Mac/ iOS development to make some repeatable operations hands-free, however sometimes things go wrong, and in perfect world I’d like to be properly informed about the errors.
+Here comes view called „Report Navigator”, which displays any type of errors, warnings at compile time of Objective-C/ Swift, or even while typing new lines of code. Let me show you how to push warnings, and errors that come out of Bash script into the Report Navigator:
+
+As usual create new `Run Script Phase`, and insert your Bash script:
+
+{% img center /images/bash-scripts-marks/bash-scripts-new-script-phase.png %}  
+
+For blog post purposes I’ve created simple script that checks which configuration is used for build:
+
+```bash
+echo "note: Starting script phase.”
+echo "warning: testing for Debug target"
+
+if [ "Debug"= "${CONFIGURATION}" ];
+then
+    echo "error: please run this script on Release target"
+else
+    echo "warning: running on Debug target"
+fi
+```
+
+Please, take a look in every echo at `note:`, `warning:` and `error:` prefixes. When XCode recognizes any of these tags, the proper indicator will show up in Report Navigator, and of course at the top bar:
+
+{% img center /images/bash-scripts-marks/bash-scripts-error-navigator.png %}  
+
+Pay attention to the colon at the end of each tag. Also the tags don’t have to be prefixes of each echo message- feel free to place it anywhere at the string.
```

Comment:
"Also the tags don’t have to be prefixes of each echo message- feel free to place it anywhere at the string." - change to something more readable